src/hooks/useChangeSelect.js

```diff
@@ -0,0 +1,43 @@
+import {useReducer, useEffect} from 'react';
+import {getChangeKey} from '../utils';
+
+const useCollection = multiple => {
+    const update = (collection, {type, value}) => {
```

Comment:
`useMemo(reducer, [multiple])`
or
`const update = (collection, {type, value, multiple}) => {`
otherwise the reducer is created every call

Reply:
Or I can just move update to outer scope